pkg/manifests/manifests.go

```diff
@@ -1659,6 +1663,33 @@ func (f *Factory) PrometheusK8s(grpcTLS *v1.Secret, trustedCABundleCM *v1.Config
 	return p, nil
 }
 
+func (f *Factory) queryLogFileConfig(p *monv1.Prometheus, queryLogFile string) (string, error) {
```

Comment:
The function can take care of setting `p.spec.queryLogFile` directly since it already modifies the p argument.

```suggestion
func (f *Factory) setupQueryLogFile(p *monv1.Prometheus, queryLogFile string) (error) {
```

in which case the caller side would look like:

```
if err != f.setupQueryLogFile(p) {
    return nil, err
}
```

and `setupQueryLogFile()` can exit early if queryLogFile is empty.